crates/iota-cluster-test/src/test_case/coin_index_test.rs

```diff
@@ -500,22 +501,16 @@ impl TestCaseImpl for CoinIndexTest {
         assert_eq!(managed_coins.len(), 40);
         assert!(managed_coins.iter().all(|c| c.balance == 5));
 
-        let mut total_coins = 0;
-        let mut cursor = None;
-        loop {
-            let page = client
+        let total_coins = GetAllPages::get_all_pages(async |cursor| {
+            client
                 .coin_read_api()
                 .get_all_coins(account, cursor, None)
                 .await
-                .unwrap();
-            total_coins += page.data.len();
-            cursor = page.next_cursor;
-            if !page.has_next_page {
-                break;
-            }
-        }
+        })
+        .await?
+        .len();
 
-        assert_eq!(iota_coins.len() + managed_coins.len(), total_coins,);
+        assert_eq!(iota_coins.len() + managed_coins.len(), total_coins);
```

Comment:
Does this pass? With `get_all_pages` `total_coins` is evaluated as the total number of pages, not the sum of coins across all pages.

Reply:
Oh I see...This is a bit confusing, indeed.

Nit: You could consider renaming to something of the sort:

* `get_all_pages -> flat_collect`
* `get_all_pages_stream -> flatten`